### PR TITLE
#1326 Run QA Suite on Deploy to Dev action

### PR DESCRIPTION
**Pull request by ldraney**
# Description

The goal of this PR is to trigger the QA suite to run against private endpoints whenever a deploy to staging or perf occurs.  As well.

issue #1326

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation changes only

## How Has This Been Tested?

How has this been tested? (e.g., screenshots, links to deployment, regression test results) to demonstrate the functionality or fix: e.g. Tested locally, Tested as a GitHub action, etc.
- [x] Successful run here:  https://github.com/department-of-veterans-affairs/notification-api/actions/runs/6151927732
with this command:  
```
gh workflow run -r 1326-qa-suite -F ref=1326-qa-suite -F environment=perf dev_deploy.yml
```
and can be tested again with the same command if you have the GH cli locally. 
Notes:
- since the last run, the name of the workflow has changed from  pull-and-run-container  to  run-QA-suite

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an appropriate title: #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/notification-api/blob/master/.github/release.yaml) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have ensured the latest master is merged into my branch and all checks are green prior to review
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] The ticket is now moved into the DEV test column

### CODE REVIEW

**Comment by cris-oddball** [comment]
Please rename this PR to `#1326 Run QA QA Suite on Deploy to Dev action`

File: bar [hunk 1]
```diff
@@ -0,0 +1 @@
+something
```
Comment (cris-oddball) [inline bar:1]: remove this file

Comment (cris-oddball) [reply bar:1]: also remove foo.bar

**Review by cris-oddball — CHANGES_REQUESTED** [review]
Please change the title of this PR to `#1326 Run QA QA Suite on Deploy to Dev action` - this is what shows up in the releases and the title should reflect the feature.

File: .github/workflows/regression-public.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr-vaec.outputs.registry }}
+        run: |
+          # Add user qa-container with UID 1051 and home directory at /home/qa-container
+          sudo useradd -u 1051 -m -d /home/qa-container -s /bin/bash qa-container
+
+          ## Set the working directory to /home/qa-container
+          cd /home/qa-container
+
+          rm -rf .env
```
[the remark this answers is not in the document]
Comment (ldraney) [reply .github/workflows/regression-public.yml:56]: you're right, i got used to working on the EC2 runner!

File: bar (hunk 1 above)
Comment (ldraney) [reply bar]: done

**Review by cris-oddball — DISMISSED** [review]

File: .github/workflows/regression-public.yml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr-vaec.outputs.registry }}
+        run: |
+          # Add user qa-container with UID 1051 and home directory at /home/qa-container
+          sudo useradd -u 1051 -m -d /home/qa-container -s /bin/bash qa-container
+
+          ## Set the working directory to /home/qa-container
+          cd /home/qa-container
+
+          #aws ssm get-parameter --name /notification-api-qa/env --with-decryption | sudo tee .env > /dev/null
```
Comment (EvanParish) [inline .github/workflows/regression-public.yml:56]: Is this commented-out code left in here intentionally?

Comment (ldraney) [reply .github/workflows/regression-public.yml]: good catch.  I've updated the comments on the file, please check again

**Review by cris-oddball — APPROVED** [review]

**Review by EvanParish — APPROVED** [review]

**Pull request merged**